olive/model/hf_onnx_config.py

```diff
@@ -0,0 +1,107 @@
+from typing import Any, Mapping, Optional, OrderedDict
+
+from transformers import PreTrainedTokenizer, TensorType
+from transformers.onnx import OnnxConfigWithPast
+
+ADDITIONAL_MODEL_TYPES = {
+    "gpt-neox": (
+        [
+            "default",
+            "default-with-past",
+            "causal-lm",
+            "causal-lm-with-past",
+            "sequence-classification",
+            "token-classification",
+        ],
+        "TextDecoderOnnxConfig",
+    ),
+    "llama": (
+        [
+            "default",
+            "default-with-past",
+            "causal-lm",
+            "causal-lm-with-past",
+            "sequence-classification",
+        ],
+        "TextDecoderOnnxConfig",
+    ),
+    "opt": (
+        [
+            "default",
+            "default-with-past",
+            "causal-lm",
+            "causal-lm-with-past",
+            "question-answering",
+            "sequence-classification",
+        ],
+        "TextDecoderOnnxConfig",
+    ),
+}
+
+
+class TextDecoderOnnxConfig(OnnxConfigWithPast):
+    # in OnnxConfigWithPast.fill_with_past_key_values_
+    # there is a bug in the name for the present sequence length dimension
+    # it should be `past_sequence` instead of `past_sequence + sequence`
+    def fill_with_past_key_values_(
+        self, inputs_or_outputs: Mapping[str, Mapping[int, str]], direction: str, inverted_values_shape: bool = False
+    ):
+        """
+        Fill the input_or_outputs mapping with past_key_values dynamic axes considering.
+
+        Args:
+            inputs_or_outputs: The mapping to fill.
+            direction: either "inputs" or "outputs", it specifies whether input_or_outputs is the input mapping or the
+                output mapping, this is important for axes naming.
+            inverted_values_shape:
+                If `True`, store values on dynamic axis 1, else on axis 2.
+
+        """
+        if direction not in ["inputs", "outputs"]:
+            raise ValueError(f'direction must either be "inputs" or "outputs", but {direction} was given')
+
+        name = "past_key_values" if direction == "inputs" else "present"
+        sequence_length_name = "past_sequence" if direction == "inputs" else "past_sequence + sequence"
+        for i in range(self.num_layers):
+            inputs_or_outputs[f"{name}.{i}.key"] = {0: "batch", 2: sequence_length_name}
```

Comment:
IIRC, kv cache may has following shapes:
1. (batch size, seq_length, hidden_size)
2. or (batch size, num_attention_head ,seq_length, hidden_size/num_attention_head ) (used in dolly/llama

Quick question if shape 1 is the case of inverted_values_shape == True?

Reply:
Yes, I think `inverted_values_shape==True` is for case 1. However, the additional models being added are for case 2 so I haven't see case 1 so far. 
The parent class `OnnxConfigWithPast` works for case 2 
https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/config.py#L530. In the future, we might have to create or inherit a new class for case 1 if needed.